File: pm_core/src/core/domain/dto/response/project_response.go
/*
Author: QuanTuanHuy
Description: Part of Serp Project
*/

package response

type ProjectResponse struct {
	ID                 int64   `json:"id"`
	TenantID           int64   `json:"tenantId"`
	Name               string  `json:"name"`
	Key                string  `json:"key"`
	Description        *string `json:"description,omitempty"`
	Status             string  `json:"status"`
	Visibility         string  `json:"visibility"`
	StartDateMs        *int64  `json:"startDateMs,omitempty"`
	TargetEndDateMs    *int64  `json:"targetEndDateMs,omitempty"`
	NextItemNumber     int     `json:"nextItemNumber"`
	TotalWorkItems     int     `json:"totalWorkItems"`
	CompletedWorkItems int     `json:"completedWorkItems"`
	CreatedAt          int64   `json:"createdAt"`
	UpdatedAt          int64   `json:"updatedAt"`
}

File: pm_core/src/main/java/serp/project/pmcore/core/domain/dto/response/ProjectResponse.java
/**
 * Author: QuanTuanHuy
 * Description: Part of Serp Project
 */

package serp.project.pmcore.core.domain.dto.response;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ProjectResponse {
    private Long id;
    private String key;
    private String name;
    private String description;
    private String url;
    private Long leadUserId;
    private Long avatarId;
    private Long categoryId;
    private String projectTypeKey;
    private Boolean isArchived;
    private Long archivedAt;

    // Scheme bindings
    private Long issueTypeSchemeId;
    private Long workflowSchemeId;
    private Long fieldConfigSchemeId;
    private Long issueTypeScreenSchemeId;
    private Long permissionSchemeId;
    private Long notificationSchemeId;
    private Long prioritySchemeId;
    private Long issueSecuritySchemeId;

    private Long createdAt;
    private Long createdBy;
    private Long updatedAt;
    private Long updatedBy;
}
